src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -184,97 +181,108 @@ object LinearMixedRegression {
       vds2
   }
 
-  def computeNEigs(S: DenseVector[Double], nEigs: Int, optDroppedVarianceFraction: Option[Double]): Int = {
-    val trace = S.toArray.sum
-    var i = S.length - 1
+  def computeNEigsDVF(S: DenseVector[Double], droppedVarianceFraction: Double): Int = {
+    require(0 <= droppedVarianceFraction && droppedVarianceFraction < 1)
+    val trace = sum(S)
+
+    var i = 0
     var runningSum = 0.0
-    if (optDroppedVarianceFraction.isEmpty) return nEigs
-    val target = optDroppedVarianceFraction.get * trace
-    while (i > nEigs || runningSum <= target) {
-      runningSum += S(-i)
-      i -= 1
+    val target = droppedVarianceFraction * trace
```

Comment:
I think it's possible (though unrealistic) that droppedVarianceFraction is close enough to 1 that target == trace and while loop throw NPE. Safest to also check that i < S.length in while loop.

You could start i = -1, move i += 1 above runningSum, and then just use S.length - i at the end without i = i - 1 or comment. You're call.